snap/snapcraft.yaml

```diff
@@ -23,8 +23,10 @@ parts:
 
       version="9999"
 
-      if echo "${GITHUB_REF:-}" | grep -q -E "^refs/tags"; then
-        version=$(echo ${GITHUB_REF:-} | cut -d/ -f3)
+      kata_url="https://github.com/kata-containers/kata-containers"
+      latest_version=$(git ls-remote --tags ${kata_url}  | egrep -o "refs.*" | egrep -v "\-alpha|\-rc|{}" | egrep -o "[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+" | sort -V -r | head -1)
+      if echo "${latest_version:-}" | grep -q -E "^refs/tags"; then
+        version=$(echo ${latest_version:-} | cut -d/ -f3)
```

Comment:
Hi @jongwu, I have a question here. IIUC, the intention is to set the correct version of kata snap packages right. Here since the value of `latest_version` would be `3.0.0` and the condition wouldn't be satisfied, wouldn't they still be set to `9999` ?
Similar to how it ran in the GitHub workflow on this PR:

```
+ version=9999
+ kata_url=https://github.com/kata-containers/kata-containers
++ git ls-remote --tags https://github.com/kata-containers/kata-containers
++ egrep -v '\-alpha|\-rc|{}'
++ egrep -o '[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+'
++ sort -V -r
++ egrep -o 'refs.*'
++ head -1
+ latest_version=3.0.0
+ grep -q -E '^refs/tags'
+ echo 3.0.0
+ snapcraftctl set-grade stable
+ snapcraftctl set-version 9999
```
Please correct me if my understanding is wrong.

Reply:
yep, I was also wondering why the conditional at the line number 28 would be needed even after `latest_version` is successfully obtained.

Reply:
Thanks @Amulyam24 @BbolroC -, that's my fault. 

What about replacing the if statement by the following:

> version="${latest_version:-$version}"

Reply:
yes, sounds good and it could directly checkout the `latest_version`.

Reply:
Okay, code is updated.

Reply:
lgtm, thanks ^^